modules/forum_pinner.js

```diff
@@ -0,0 +1,78 @@
+const { SlashCommandBuilder } = require("@discordjs/builders");
+
+async function pinPost(thread) {
+  let messages = await thread.messages.fetch();
```

Comment:
Most of this can be ignored if the /pin command is staff only, but does this have any consequences if ran on a very long (>1k messages) thread? I know we have a few threads with many hundreds of messages.

Interactions have a short time frame for their initial response, but are allowed to defer it for slightly longer. However, if this command were to be ran multiple times, on different threads or even the same thread, message fetch rate limits might be hit very quickly. A [queue](https://www.npmjs.com/package/p-queue) for pinning might be appropriate, and possibly even some kind of check (even if just in-memory) to see if a thread has already been pinPost'd.

Reply:
I didn't have enough data for testing this command thoroughly, but it's made for threads made prior to the module running on the bot.
It tries to get the first message and pins it, but may fail.
Since it can only be ran by staff members, has to be ran rarely (mainly when the bot is down), and staff members have been pinning the messages without the post, this command may not even be needed at all.